src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -427,6 +446,159 @@ void runtime::intelgpu::do_max_pool_backprop_operation(cldnn::topology& topology
     topology.add(op_max_pool_backprop);
 }
 
+void runtime::intelgpu::do_avg_pool_backprop_operation(cldnn::topology& topology,
+                                                       const string& delta_name,
+                                                       const Shape& delta_shape,
+                                                       const string& output_name,
+                                                       const Shape& output_shape,
+                                                       const element::Type& output_type,
+                                                       const Shape& win_shape,
+                                                       const Shape& win_stride,
+                                                       const Shape& pad_below,
+                                                       const bool include_padding)
+{
+    const string entry_point_name = "op_avg_pool_backprop_" + output_name;
+    codegen::CodeWriter writer;
+    vector<size_t> gws;
+
+    const Shape delta_data(delta_shape.cbegin() + 2, delta_shape.cend());
```

Comment:
Could you please check the size of the Shape before this construction?
Please use `static Shape get_channel_shape(const Shape& shape, const string& function_name)` as an example from `intelgpu_batch_norm.cpp`. Let's make this function a bit more general by adding `size_t min_size` parameter for the Shape limit. 
Also, please add these checks to `do_max_pool_backprop_operation` - I forgot to add them previously.

Reply:
Shape of delta is checked in the definition of the operation, so is it necessary? (https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/avg_pool.cpp#L235 )